src/core/geometry/qgspolygon.cpp

```diff
@@ -185,6 +185,50 @@ QByteArray QgsPolygon::asWkb( QgsAbstractGeometry::WkbFlags flags ) const
   return wkbArray;
 }
 
+QString QgsPolygon::asWkt( int precision ) const
+{
+  QString wkt = wktTypeStr();
+
+  if ( isEmpty() )
+    wkt += QLatin1String( " EMPTY" );
+  else
+  {
+    wkt += QLatin1String( " (" );
+    if ( mExteriorRing )
+    {
+      QString childWkt = mExteriorRing->asWkt( precision );
+      if ( qgsgeometry_cast<QgsLineString *>( mExteriorRing.get() ) )
+      {
+        // Type names of linear geometries are omitted
+        childWkt = childWkt.mid( childWkt.indexOf( '(' ) );
+      }
+      wkt += childWkt + ',';
+    }
+    for ( const QgsCurve *curve : mInteriorRings )
+    {
+      QString childWkt;
+      if ( true || curve->hasCurvedSegments() )
```

Comment:
Hmm, what's the actual issue here? Is it a bad geometry with curved rings in a polygon type?

Reply:
Yes, actually I've just found a couple of duplicated issues:  https://github.com/qgis/QGIS/issues/50869 and probably a better fix in https://github.com/qgis/QGIS/pull/55371

I am in favor of closing this if #55371 fixes this issue.

Reply:
We are even expecting invalid WKT in the test: https://github.com/qgis/QGIS/blob/master/tests/src/app/testqgsmaptooladdring.cpp#L166 , that should not have been accepted in the first place.

Reply:
@nyalldawson I thought about this, IMO the best course of action is to fix the wrong geometry generation in the first place (i.e. fix the maptool or the editing utils which generates the wrong geometry) but also to leave the WKT export cleaning code, because the "perfect" solution which is rewrite the geometry classes in a way that makes it impossible to construct an invalid POLYGON implies a deep refactoring and its way out of scope for this bug fixing effort. And because we don't know how many times and where exactly this wrong geometries can be generated by the QGIS application or by other code which uses the QGIS API.

Reply:
@elpaso I agree with you. Having worked on it extensively, I've initiated several PRs to address these issues, but there's still a lot to be done. The challenge lies in QGIS being quite lenient on input; it's not necessarily a problem, as users tend to complain otherwise. However, ensuring the accuracy of exports is crucial. I've started working on these matters, but it's still a work in progress.

Reply:
@lbartoletti since you know very well the topic and the issues can you please have a look to this PR?